[JOBS] New ReactOS Website: Redo the Jobs page. ONLINE-822 (#17)
reactos/web-content: content/jobs.md

The ReactOS Project is seeking qualified individuals to fill multiple positions. Contracted **full time, part time, and scholarship (internship)** level positions are all available.

> If you do qualify for a position, you'll be joining the very small and exclusive ranks of people who know how to do NT systems development. And believe us, those ranks are very, very small. And yes, you will be gaining an immense amount of knowledge working with such people. 

Reviewer: https://jira.reactos.org/browse/ONLINE-822 states that we want to make us seem less elitist.
This sentence goes directly against that.

Author: We could maybe change the "seeking qualified individuals" to "seeking individuals" and

" If you do qualify for a position, you'll be joining the very small and exclusive ranks of people who know how to do NT systems development. And believe us, those ranks are very, very small. And yes, you will be gaining an immense amount of knowledge working with such people. "

to

"If you qualify for a position, you'll be joining the very small and exclusive ranks of people who know how to do NT systems development. And believe us, those ranks are very, very small so you will be gaining an immense amount of knowledge working with such people. It's a great way for beginners to interact with people who have been working in this field for over the years."

This way it looks more beginner-friendly.

Available Positions
===

Developer
---

> We are looking for people proficient in C and its a big plus if you are proficient in C++ too.C++ proficiency does not equate C proficiency so do not assume you have the latter if you have the former. Also keep in mind, again, that these are the minimum requirements. On top of these you need to be a good problem solver, belief in the mission of ReactOS, and most importantly, be able to demonstrate that you can work with the ReactOS codebase via a few patches.

Reviewer: Nit: space after `.`.
Furthermore: explaining a difference between c and c++ is good, but for some positions c++ is more important,
other c is just fine.
(f.e. drivers / kernel mode stuff etc are generally c code, the shell and usermode applications lean more toward c++)

Author: Okay, I will be elaborating a bit about the C and C++ requirement part.

Author: I've made changes to the developer section. Please do have a look at it.

[Development Introduction](https://reactos.org/wiki/Development_Introduction)

Every year we participate in [Google Summer of code](https://summerofcode.withgoogle.com/) which is a program by Google which lets you work with us during the summer for fixing one of the major bugs. You get paid a certain amount of money for the work. 

We even conduct hackfests every year where our team comes together and sit together to fix issues.

Tester
---
No operating system is usable if it is found to be unstable and prone to crashes. As ReactOS development work involves studying the behavior of an operating system that is not completely documented, testing should be of even greater importance in order to fulfill the objective of binary compatibility with Windows.

> You can assist the ReactOS development effort by installing regular trunk builds available here and providing feedback on issues/problems you encountered during and after installing the OS via our bug tracker . More information with regard to debugging is available here and is recommended reading for those who wish to submit a report to our task tracker.For this, you need to have basic knowledge in collecting information from a crash like backtraces and crash dumps. Knowledge of using a basic debugger is a plus.

Reviewer: trunk = master.
>More information with regard to debugging is available here 
'here' probably should link to something

Author: I've linked the wiki page down there. It's linked as "getting started". If needed will add a hyperlink to the setting up a debugging environment wiki page.

We have wiki pages for setting up a debugging environment and building using different compilers too.

[Getting Started](https://reactos.org/wiki/Welcome_to_the_ReactOS_Development_Wiki)

[How to create a JIRA Issue](https://reactos.org/wiki/User:Oldman/A_layman%27s_guide/How_to_create_a_JIRA_Issue)

Jira maintainer
---

> We use JIRA for keeping track of issues. This position is for people interested in maintaining the JIRA. You will have to look after the development/work progress in the issues and mark them resolved. Currently, we have a system that links each of the PR from GitHub to the related issue in JIRA.In this role, you will mainly be working to make sure everything in JIRA works as expected.

Reviewer: ```suggestion
We use JIRA to keep track of issues. This position is for people interested in maintaining JIRA. You will have to look after the development/work progress in the bugtracker and assist in triaging the issues. This includes but is not limited to: Asking for the exact version that was tested, disseminating logs (or asking further steps to provide those), labeling issues. Currently, we have a system that links each of the PR from GitHub to the related issue in JIRA. In this role, you will mainly be working to make sure everything in JIRA works as expected.
```

Author: I did not have any clues on what a JIRA maintainer has to do. So I will be directly merging your suggestion. Thanks for the edit.

[JIRA](https://jira.reactos.org/)

Content editor / creator
---
We are a small team of individuals working, so we need some creative people to look after the posts(content to be posted in our blog) and our site. In this role, you will need to write articles, make changes or update the wiki pages. This could be adding new information or images. You will mainly be working on this site and the wiki pages. Our current site is generated using [Hugo](https://gohugo.io/). New design ideas are welcome too.

[Wikipage](https://reactos.org/wiki/Welcome_to_the_ReactOS_Development_Wiki)

Community manager
---

> We have an IRC channel,a forum and a mattermost channel. The role is to keep people involved, keep discussions going and moderate.

Reviewer: ```suggestion
We have a [mattermost instance](chat.reactos.org), [a forum](reactos.org/forum) and an IRC channel (#reactos on freenode). The role is to keep people involved, keep discussions going and moderate.
```

Author: Thanks for the cleaner edit :)

The ReactOS Project is seeking qualified individuals to fill multiple positions. Contracted **full time, part time, and scholarship (internship)** level positions are all available.

If you qualify for a position, you'll be joining the very small and exclusive ranks of people who know how to do NT systems development. And believe us, those ranks are very, very small so you will be gaining an immense amount of knowledge working with such people. It's a great way for beginners to interact with people who have been working in this field for over the years.

Available Positions
===

Developer
---

> We are looking for people proficient in C and its a big plus if you are proficient in C++ too.C++ proficiency does not equate C proficiency so do not assume you have the latter if you have the former. Also keep in mind, again, that these are the minimum requirements. On top of these you need to be a good problem solver, belief in the mission of ReactOS, and most importantly, be able to demonstrate that you can work with the ReactOS codebase via a few patches.

Reviewer: ```suggestion
We are looking for people proficient in C and its a big plus if you are proficient in C++ too. C++ proficiency does not equate C proficiency so do not assume you have the latter if you have the former. Also keep in mind, again, that these are the minimum requirements. On top of these you need to be a good problem solver, belief in the mission of ReactOS, and most importantly, be able to demonstrate that you can work with the ReactOS codebase via a few patches / pull requests.
```

> In ReactOS, we need all kinds of people, those who are ready to port drivers and mainly work in the kernel region and those of you who would like to work in user-mode.For all the work related to kernel and drivers, you need to be proficient in C and don't need C++ for most of it. But when it comes to user-mode, you do need to have knowledge of C++. As an example, most of the applications are written in C++ not in C.Whereas when we have a look at the drivers we notice that they are written in C.

Reviewer: On https://master.web-content.reactos.org/developing/ (temp url) there is a rough explanation about the types of developers / development work

Author: Rather than explaining it all here in the positions page wouldn't it be better just to link it up? @learn-more.
We could maybe add a line that says "For detailed information on what we do, refer -[Development Information](https://master.web-content.reactos.org/developing/)". What do you say about this?

Author: > In ReactOS, we need all kinds of people, those who are ready to port drivers and mainly work in the kernel region and those of you who would like to work in user-mode. For all the work related to kernel and drivers, you need to be proficient in C and don't need C++ for most of it. But when it comes to user-mode, you do need to have knowledge of C++. As an example, most of the applications are written in C++ not in C.Whereas when we have a look at the drivers we notice that they are written in C.

would be changed to 

> In ReactOS, we need all kinds of people like, those who are ready to port drivers and mainly work in the kernel region, those of you who would like to **work in different other things such as System Processes, Shell and Explorer and Win32 API. If you would like to have a detailed look at what we do, you can refer to our [page](https://master.web-content.reactos.org/developing/) which explains it all in detail.** For all the work related to kernel and drivers, you need to be proficient in C and don't need C++ for most of it. But when it comes to user-mode, you do need to have knowledge of C++. As an example, most of the applications are written in C++ not in C.Whereas when we have a look at the drivers we notice that they are written in C.

Reviewer: yeah seems good

Author: Cool will finish that part up :)

> In ReactOS, we need all kinds of people like, those who are ready to port drivers and mainly work in the kernel region, those of you who would like to work in different other things such as System Processes, Shell and Explorer and Win32 API. If you would like to have a detailed look at what we do, you can refer to our [page](/developing) which explains it all in detail. For all the work related to kernel and drivers, you need to be proficient in C and don't need C++ for most of it. But when it comes to user-mode, you do need to have knowledge of C++. As an example, most of the applications are written in C++ not in C.Whereas when we h  ave a look at the drivers we notice that they are written in C.

Reviewer: ```suggestion
In ReactOS, we need all kinds of people: those who are ready to port drivers and/or work mainly in the kernel region, those who would like to work in different other things such as Shell, Explorer and Win32 API. If you would like to have a detailed look at what we do, you can refer to our [page](/developing) which explains it all in detail. For all the work related to kernel and drivers, you need to be proficient in C. But when it comes to user-mode, you do need to have knowledge of / experience with C++. As an example, most of the applications are written in C++ not in C. Whereas when the drivers / kernel mode components are mainly written in C.
```

For more information on the C++ part, you can have a look at [Win32 development using C++](https://docs.microsoft.com/en-us/windows/win32/LearnWin32/learn-to-program-for-windows).

> For those of you interested can have a look at our wiki page which pretty much covers all of the details on getting started with development for ReactOS.

Reviewer: ```suggestion
For those of you interested, you can have a look at our wiki page which pretty much covers all of the details on getting started with development for ReactOS.
```

[Development Introduction](https://reactos.org/wiki/Development_Introduction)

> Every year we participate in [Google Summer of code](https://summerofcode.withgoogle.com/) which is a program by Google which lets you work with us during the summer for fixing one of the major bugs. You get paid a certain amount of money for the work. 

Reviewer: ```suggestion
Every year we participate in [Google Summer of code](https://summerofcode.withgoogle.com/) which is a program by Google which lets you work with us during the summer to work on a major project. Google pays the student, and ReactOS provides guidance / assistance in the form of a mentor. 
```

We even conduct hackfests every year where our team comes together and sit together to fix issues.

Tester
---

> No operating system is usable if it is found to be unstable and prone to crashes. As ReactOS development work involves studying the behavior of an operating system that is not completely documented, testing should be of even greater importance in order to fulfill the objective of binary compatibility with Windows.

Reviewer: ```suggestion
No operating system is usable if it is found to be unstable and prone to crashes. As ReactOS development work involves studying the behavior of an operating system that is not completely documented, testing is of even greater importance in order to fulfill the objective of binary compatibility with Windows.
```

> You can assist the ReactOS development effort by installing regular trunk(master) builds available here and providing feedback on issues/problems you encountered during and after installing the OS via our bug tracker . More information with regard to debugging is available here and is recommended reading for those who wish to submit a report to our task tracker.For this, you need to have basic knowledge in collecting information from a crash like backtraces and crash dumps. Knowledge of using a basic debugger is a plus.

Reviewer: ```suggestion
You can assist the ReactOS development effort by installing regular nightly (master) builds available [here](https://reactos.org/getbuilds/) and providing feedback on issues/problems you encountered during and after installing the OS via our [bug tracker](https://jira.reactos.org/). More information with regard to debugging is available [here](https://reactos.org/wiki/Debugging) and is recommended reading for those who wish to submit a report to our bug tracker. For this, you need to have basic knowledge in collecting information from a crash like backtraces and crash dumps. Knowledge of using a basic debugger is a plus.
```
